WS2-1606 - Articles: Added media/hero, updated node edit form and lay…

bjdevil21 opened the pull request:
…out/styling

First attempted PR on this repo.

mlsamuelson commented:
[![Visit Site](https://raw.githubusercontent.com/pantheon-systems/terminus-build-tools-plugin/master/assets/images/visit-site-36.png)](https://pr-5-webspark-ci.pantheonsite.io/)

Created multidev environment [pr-5](https://dashboard.pantheon.io/sites/aec2f75f-35eb-41f8-abe7-95b292415259#pr-5) for webspark-ci.

mlsamuelson reviewed (changes requested):
See comments with wording changes for improved UX requested.

Also, note when there is no breadcrumb, the heading padding looks off. 

![Screenshot 2023-05-10 at 7 46 24 AM](https://github.com/ASUWebPlatforms/webspark-ci/assets/212131/c886b47d-33a2-4d73-a4b4-3561cff8fd0c)

Am I correct in assuming this is based off the https://unity.web.asu.edu/@asu/bootstrap4-theme/index.html?path=/story/molecules-heroes-examples--story-hero-large&args=template:0;header:false;footer:false story? Does the solve need to be implemented in UDS?

Lastly, this issue was reported as a regression. Are the fields and ymls updates added adding back in missing items? Has consideration and testing been done on how those transition? In previous testing with missing yml files for fields in the last release - when they are added back in (if the fields have the same name and storage specs) the previous values re-emerge.    Looks like these yml files for the fields have both the `field.field.node.*` and `field.storage.node.*` variants in the branch - so perhaps the issue was only some of them went missing? Just looking for clarification to ensure this resolves the issue fully.

In sum:
1. Changes requested to wording per other comments).
2. Seeking clarification on if header spacing issue is a WS2 or UDS problem to solve. If WS2, it should be addressed in this PR.
3. Need answer to how this will roll out as an update.

bjdevil21 commented on web/profiles/webspark/webspark/config/install/field.field.node.article.field_author.yml:
```diff
@@ -8,8 +8,8 @@ id: node.article.field_author
-label: Author
-description: ''
+label: 'Article author'
+description: 'Author of the article (not the Drupal page creator). Will be displayed above article.'
```
Done.

bjdevil21 commented on web/profiles/webspark/webspark/config/install/field.field.node.article.body.yml:
```diff
@@ -9,17 +9,14 @@ dependencies:
 label: Body
-description: ''
-required: false
+description: 'The body of the article goes here. Include any media, images, or other elements here (not as a separate Drupal block).'
```
Done.

bjdevil21 commented on web/profiles/webspark/webspark/webspark.install:
```diff
@@ -277,3 +277,23 @@ function webspark_update_9012(&$sandbox) {
+  \Drupal::service('webspark.config_manager')->updateConfigFile('core.entity_view_display.node.article.default');
+  // Lock the configuration storage.
+  \Drupal::state()->set('configuration_locked', TRUE);
+  \Drupal::logger("webspark")->notice("Updates to Article content type were attempted.");
```
The "updates were attempted" was verbiage I'd seen over the years in Drupal updates that isn't great. In the past (D5-D7), it's really meant that "updates were attempted and no errors were caught and output in this log". In my limited experience with D8+, it's seemed like errors are better caught these days. 

I've applied your "Article content type updated" suggestion.

bjdevil21 commented:
1. These good suggestions are added and pushed up.
2. The header spacing issue was addressed in the web/themes/webspark/renovation/src/sass/content/_node.article.scss file.  but not in the CSS file (only locally tested). After your refresher answer today on the CSS/JS question, I've updated the assets directory with the compiled update.
3. There were no previously deleted and re-added field.field.* fields in this PR or in WS2-1167 (where I initially added the new fields).

The reported regression was due to the tech.asu.edu site builders modifying the Article content type to their needs. This fix will only partially bring them back into alignment, but I've laid out specific steps for Garrett M. to apply the updates in a special manner to fix it for them. (That's another discussion, but in short, they broke a cardinal rule: Don't hack Webspark core.)

I did forget the field.field.* changes initially in this PR but I caught the issue and updated this PR before your last review.

bjdevil21 commented:
I just fixed a conflict related to hook_update_N number collisions from another merged ticket (moved 9013 to 9014).

bjdevil21 commented:
This is ready for another review over the three questions @mlsamuelson presented above.
